docs/OV_Runtime_UG/supported_plugins/GPU.md

```diff
@@ -258,7 +288,51 @@ The code snippet below demonstrates how to use dynamic batching in simple scenar
          :fragment: dynamic_batch
 
 
-For more details, see the :doc:`dynamic shapes guide<openvino_docs_OV_UG_DynamicShapes>`.
+Notes for performance and memory consumption in dynamic shape
+-----------------------------------------------------------
+
+- Extra CPU utilization during inference :
+
+   - Shape inference for new input shapes
+   - Kernel compilation in runtime for optimal kernel
+   - Unfusion of the fused subgraph when the fusing is illegal for runtime shape
+
+-  Higher memory consumption for in-memory cache
+
+   - Optimal kernels and weights from the previously used shapes are preserved in in-memory cache for future use
+
+
+Recommendations for performance improvement
+-----------------------------------------------------------
+
+- Use static shape whenever possible
+
+   - Static models can benefit from more agressive optimizations such as constant propagation, fusing, reorder optimizatio.
+   If the same shape is used for dynamic model, the performance is worse than running that shape with static model.
+   It is therefore recommended to reshape the dynamic model to a static model, if the scenario allows.
+
+- Use bounded dynamic shape whenever possible
+
+   - The GPU plugin needs to reallocate memory if the current shape is larger than the maximum of the previous shapes, which causes additional overhead.
+   - Using bounded dynamic shape will help reduce such overhead. For example, use {ov::Dimension(1, 10), ov::Dimension(1, 384)} instead of {ov::Dimension(-1), ov::Dimension(-1)}.
+   - Note that bounded dynamic *batch* is handled differently as mentioned above.
+
+- Use permanent cache, i.e., either OpenCL cache or OpenVino model_cache, to reduce runtime re-compilation overhead
+
+   - GPU plugin deploys in-memory cache to store compiled kernels for previously used shapes, but the size of such in-memory cache is limited.
+   Therefore, it is recommended to use either of the following permanent caches too:
+
+      - OpenVino model_cache: See the :doc:`Model caching overview<openvino_docs_OV_UG_Model_caching_overview>`
+      - OpenCL cache : See the page : https://github.com/intel/compute-runtime/blob/master/opencl/doc/FAQ.md#feature-cl_cache
+
+- The longer the inference sequence, the better throughput can be obtained, because it can leverage the runtime compilation time.
+
+   - If the primitive has a shape-agnostic kernel, it uses that kernel for the first inference.
```

Comment:
`the first inference` may be misleading here. I think it's better to say that shape agnostic kernel is used whenever static implementation doesn't exist in in-memory kernels cache for current tensor shapes

Reply:
Fixed